aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1650,6 +1685,14 @@ Tensor transpose(const Tensor & self, int64_t dim0, int64_t dim1) {
   auto ndims = self.dim();
   dim0 = maybe_wrap_dim(dim0, ndims);
   dim1 = maybe_wrap_dim(dim1, ndims);
+
+  if (self.is_sparse_csr()) {
+    // Sparse CSR transpose is a copy operation as the values of
+    // transposed CSR tensor are permuted values of the input CSR
+    // tensor.
```

Comment:
NIT: You could add this to the other sparse branch as well or unify them and then dispatch to COO or CSR as needed.

Reply:
Done.